modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/OAuthOperationMessageProcessor.java

```diff
@@ -96,23 +99,45 @@ public void error(Throwable e) {
           return;
         }
 
-        if (LOGGER.isDebugEnabled()) {
-          LOGGER
-              .debug(format("AccessToken for resourceOwner '%s' expired while executing operation '%s:%s' using config '%s'. "
-                  + "Will attempt to refresh token and retry operation", connectionProvider.getResourceOwnerId(),
-                            getExtensionModel().getName(), operationContext.getComponentModel().getName(),
-                            operationContext.getConfiguration().get().getName()));
-        }
+        Reference<Optional<String>> resourceOwnerIdReference = new Reference<>(empty());
+        connectionProvider.getGrantType().accept(new OAuthGrantTypeVisitor() {
+
+          @Override
+          public void visit(AuthorizationCodeGrantType grantType) {
+            AuthorizationCodeConnectionProviderWrapper cp = (AuthorizationCodeConnectionProviderWrapper) connectionProvider;
+            String rsId = cp.getResourceOwnerId();
+            resourceOwnerIdReference.set(of(rsId));
+
+            if (LOGGER.isDebugEnabled()) {
+              LOGGER.debug(format(
+                                  "AccessToken for resourceOwner '%s' expired while executing operation '%s:%s' using config '%s'. "
+                                      + "Will attempt to refresh token and retry operation",
+                                  rsId, getExtensionModel().getName(), operationContext.getComponentModel().getName(),
+                                  operationContext.getConfiguration().get().getName()));
+            }
+          }
+
+          @Override
+          public void visit(ClientCredentialsGrantType grantType) {
+            if (LOGGER.isDebugEnabled()) {
+              LOGGER.debug(format(
+                                  "AccessToken for expired while executing operation '%s:%s' using config '%s'. "
+                                      + "Will attempt to refresh token and retry operation",
+                                  getExtensionModel().getName(), operationContext.getComponentModel().getName(),
+                                  operationContext.getConfiguration().get().getName()));
+            }
+          }
+        });
+
+        Optional<String> resourceOwnerId = resourceOwnerIdReference.get();
 
-        String ownerConfigName = operationContext.getConfiguration().get().getName();
         try {
-          oauthManager.refreshToken(ownerConfigName, expiredException.getResourceOwnerId(),
-                                    getOAuthConnectionProvider(operationContext));
+          connectionProvider.refreshToken(resourceOwnerId.orElse(""));
```

Comment:
should,t you fail in this case when notPresent?

Reply:
nope. The concept doesn't apply to all grant types